### PR TITLE
Create new gui window

### PR DESCRIPTION
**Pull request by d3m0**
Changed GUI.
Added reverseList method.
Added Top5 riders

### CODE REVIEW

File: src/main/java/com/d3m0/ts/core/Rider.java [hunk 1]
```diff
@@ -3,7 +3,7 @@
 /**
  * Created by d3m0 on 24.10.2015.
  */
-public class Rider {
+public class Rider implements Comparable {
```
Comment (pldpv) [inline src/main/java/com/d3m0/ts/core/Rider.java]: Please use Comparable<Rider>

File: src/main/java/com/d3m0/ts/core/Rider.java [hunk 2]
```diff
@@ -47,4 +47,19 @@ public String getNumber() {
     public void setNumber(String number) {
         this.number = number;
     }
+
+    @Override
+    public int compareTo(Object o) {
+        Double timeOnTrack = ((Rider) o).getTimeOnTrack();
+
+        if (this.timeOnTrack > timeOnTrack) {
```
Comment (IevgenTkachuk) [inline src/main/java/com/d3m0/ts/core/Rider.java]: use compare method from Double class

File: src/main/java/com/d3m0/ts/gui/MainWindow.java [hunk 3]
```diff
@@ -50,48 +54,126 @@ public void actionPerformed(ActionEvent e) {
             @Override
             public void actionPerformed(ActionEvent e) {
                 timer.stop();
-                race.setDefaultNumber(numberByDefault.getText());
+                race.setDefaultNumber(numberByDefaultInput.getText());
                 race.stopTime(new UsbAction());
                 showResults(race.getRidersOnTrack());
-                race.removeFinishedRider();
                 totalRidersOnTrackLabel.setText(String.valueOf(race.getRidersOnTrack().size()));
                 if (race.getRidersOnTrack().size() == 0) {
```
Comment (IevgenTkachuk) [inline src/main/java/com/d3m0/ts/gui/MainWindow.java]: use isEmpty()

File: src/main/java/com/d3m0/ts/gui/MainWindow.java [hunk 4]
```diff
@@ -50,48 +54,126 @@ public void actionPerformed(ActionEvent e) {
...
+            elements++;
+        }
+
+        top5Label.setText(output);
+    }
+
+    private List<Rider> reverseList(Map<String, Rider> ridersMap) {
+        List<Rider> riders = new ArrayList<>(ridersMap.values());
+
+        Stack<Rider> stack = new Stack<Rider>();
```
Comment (IevgenTkachuk) [inline src/main/java/com/d3m0/ts/gui/MainWindow.java]: Use Deque interface instead of Stack

File: src/main/java/com/d3m0/ts/core/Rider.java (hunk 1 above)
Comment (d3m0) [inline src/main/java/com/d3m0/ts/core/Rider.java]: Fixed

File: src/main/java/com/d3m0/ts/core/Rider.java (hunk 2 above)
Comment (d3m0) [inline src/main/java/com/d3m0/ts/core/Rider.java]: Fixed

File: src/main/java/com/d3m0/ts/gui/MainWindow.java (hunk 3 above)
Comment (d3m0) [inline src/main/java/com/d3m0/ts/gui/MainWindow.java]: Fixed

File: src/main/java/com/d3m0/ts/gui/MainWindow.java (hunk 4 above)
Comment (d3m0) [inline src/main/java/com/d3m0/ts/gui/MainWindow.java]: Fixed

**Pull request merged**